src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,113 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (movie: Movie) => void;
+}
+
+const isValidUrl = (url: string): boolean => {
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;// eslint-disable-line
+
+  return pattern.test(url);
+};
+
+export const NewMovie: React.FC<NewMovieProps> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [formData, setFormData] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const [allFieldsFilled, setAllFieldsFilled] = useState(false);
+
+  useEffect(() => {
+    setAllFieldsFilled(
+      formData.title.trim() !== ''
+      && formData.imgUrl.trim() !== ''
+      && formData.imdbUrl.trim() !== ''
+      && formData.imdbId.trim() !== '',
+    );
+  }, [formData]);
```

Comment:
```suggestion

      const allFieldsFilled = formData.title.trim()
      && formData.imgUrl.trim()
      && formData.imdbUrl.trim()
      && formData.imdbId.trim();
```
it is redundant to manage state for allFieldsFilled. enough to write in constant. and expression can be shorter.